### ISSUE TITLE
Image Update changes more files than it should

### ISSUE THREAD

**Issue by nomeelnoj** [fluxcd/image-automation-controller#123]
When running the image update automation, it updates ALL the files with that image policy, not just the ones that are deployed in the cluster.  For example, we have a pod called `admin`, and we deploy it in dev, stg, and prd.  Rather than have to manage multiple image policy objects, we create a single image policy called `admin`, with different policy rules as kustomize overlays for each environment.  However, when one environment gets an update, Flux commits back and updates ALL the files.  Here is an example:

```
# Flux Config
apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
kind: Kustomization
metadata:
  name: flux-system
  namespace: flux-system
spec:
  interval: 10m0s
  path: ./deploy/dev
  prune: true
  sourceRef:
    kind: GitRepository
    name: flux-system
  validation: client
```
The kustomization in question is:
```
apiVersion: kustomize.toolkit.fluxcd.io/v1beta1
kind: Kustomization
metadata:
  name: admin
  namespace: flux-system
spec:
  interval: 10m0s
  path: ./resources/admin/overlays/dev
  prune: true
  sourceRef:
    kind: GitRepository
    name: flux-system
  validation: client
```

```
# Image policy for dev
apiVersion: image.toolkit.fluxcd.io/v1alpha1
kind: ImagePolicy
metadata:
  name: admin
  namespace: flux-system
spec:
  filterTags:
    extract: $1
    pattern: ^development-v(.*)
  imageRepositoryRef:
    name: admin
  policy:
    semver:
      range: 2.x.x-0
```

```
# Image policy for stg
apiVersion: image.toolkit.fluxcd.io/v1alpha1
kind: ImagePolicy
metadata:
  name: admin
  namespace: flux-system
spec:
  filterTags:
    extract: $1
    pattern: ^staging-v(.*)
  imageRepositoryRef:
    name: admin
  policy:
    semver:
      range: 2.x.x-0
```

Then the workloads for dev/stg/prd have the same image policy, albeit different images:
```
image: myrepo/admin:development-v2.150.0-3046 # {"$imagepolicy": "flux-system:admin"}
image: myrepo/admin:staging-v2.146.0-98 # {"$imagepolicy": "flux-system:admin"}
image: myrepo/admin:production-v2.136.0-97 # {"$imagepolicy": "flux-system:admin"}
```

The issue comes when the development image is updated to `development-v2.151.0-3047`, as here is snippet of our recent commit from flux:

```diff
-image: myrepo/admin:development-v2.150.1-3046 # {"$imagepolicy": "flux-system:admin"}
-image: myrepo/admin:staging-v2.146.0-98 # {"$imagepolicy": "flux-system:admin"}
-image: myrepo/admin:production-v2.136.0-97 # {"$imagepolicy": "flux-system:admin"}
+image: myrepo/admin:development-v2.151.0-3047 # {"$imagepolicy": "flux-system:admin"}
+image: myrepo/admin:development-v2.151.0-3047 # {"$imagepolicy": "flux-system:admin"}
+image: myrepo/admin:development-v2.151.0-3047 # {"$imagepolicy": "flux-system:admin"}
```

Am I modeling this incorrectly? I am surprised that flux is configured to update files that it is not supposed to be looking at for deployment.  This is currently blocking us from moving forward with deploying into stg and prod, as we definitely cannot have staging and prod update whenever an image is pushed to dev.  My short term solution will be to modify the names of each policy, but that seems like an anti pattern.  Any help would be very much appreciated!

**Comment by darksinge** [comment]
I was just running into this problem yesterday and found (this issue)[#100]. Apparently, with the current implementation, that's simply how the automation controller works.

**Comment by stefanprodan** [comment]
Fixed in #100
